crates/optimism/txpool/src/interop.rs

```diff
@@ -0,0 +1,40 @@
+//! Additional support for pooled transactions with [`TransactionInterop`]
+
+/// Helper trait that allows attaching a [`TransactionInterop`].
+pub trait MaybeInteropTransaction {
+    /// Attach a [`TransactionInterop`].
+    fn set_interop(&mut self, conditional: TransactionInterop);
+
+    /// Get attached [`TransactionInterop`] if any.
+    fn interop(&self) -> Option<TransactionInterop>;
+
+    /// Helper that sets the conditional and returns the instance again
+    fn with_interop(mut self, interop: TransactionInterop) -> Self
+    where
+        Self: Sized,
+    {
+        self.set_interop(interop);
+        self
+    }
+}
+
+/// Helper to keep track of cross transaction interop validity
+#[derive(Debug, Clone, Default, Eq, PartialEq)]
+pub struct TransactionInterop {
+    /// Unix timestamp until which tx if considered valid by supervisor.
+    ///
+    /// If None - tx is not validated, it should be automatically revalidated by interop tracker.
+    pub timeout: Option<u64>,
```

Comment:
we can remove the option here because we assume all interops need a timeout?

Reply:
I want to keep this for the case when we didn't validation this tx with interop even once

Reply:
But we could use 0, if we haven't check it yet, for example